### PR TITLE
Improve custom commands interface

### PR DESCRIPTION
**Pull request by Mihai22125**
Allow a name to be specified in the prompt so that prompts can be referred by name.
```
- key : 'N'
    description: 'create annotated tag'
    command: "git tag -a {{ .Form.tagName }} -m \"{{ .Form.tagMessage }}\""
    context: 'tags'
    prompts:
      - type: 'input'
        title: 'Annotated tag name:'
        key: 'tagName'
      - type: 'input'
        title: 'Annotated tag message:'
        key: 'tagMessage'
```

Referring to prompt responses by index is kept in the code for backwards compatibility.

Add functionality for returning an error if the referred prompt is not defined.

This resolves: #2103

### CODE REVIEW

**Comment by jesseduffield** [comment]
Code looks good, could you add an integration test for this? You can use https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/tests/custom_commands/multiple_prompts.go as a reference. Also check out https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md

Lemme know if you need any guidance

**Comment by Mihai22125** [comment]
Hi @jesseduffield, I would be happy to give it a shot.

File: pkg/integration/tests/custom_commands/form_prompts.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	Skip:         false,
+	SetupRepo: func(shell *Shell) {
+		shell.EmptyCommit("blah")
+	},
+	SetupConfig: func(cfg *config.AppConfig) {
+		cfg.UserConfig.CustomCommands = []config.CustomCommand{
+			{
+				Key:     "a",
+				Context: "files",
+				Command: `echo "{{index .Form.FileContent}}" > {{index .Form.FileName}}`,
```
Comment (jesseduffield) [inline pkg/integration/tests/custom_commands/form_prompts.go:20]: is the `index` thing here necessary?

Comment (Mihai22125) [reply pkg/integration/tests/custom_commands/form_prompts.go]: It looks like I forgot to remove it. I updated the code now

**Comment by jesseduffield** [comment]
Thanks for waiting for my re-review. Great work!

**Comment by Mihai22125** [comment]
I'm glad that I could contribute to this cool project!

**Pull request merged**